heron/packing/src/java/org/apache/heron/packing/builder/ResourceRequirement.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.heron.packing.builder;
+
+import java.util.Objects;
+
+import org.apache.heron.common.basics.ByteAmount;
+
+/**
+ * Helper class that captures the RAM and CPU requirements of each component
+ */
+public class ResourceRequirement implements Comparable<ResourceRequirement> {
+
+  private String componentName;
+  private ByteAmount ramRequirement;
+  private double cpuRequirement;
+  private SortingStrategy sortingStrategy;
```

Comment:
SortingStrategy is a bit strange to be here I feel. Does it make more sense to refactor it out?

Reply:
Refactored so that now `SortingStrategy` becomes a comparator itself.